### PR TITLE
Fixes #9919: use suse_version instead of sles_version during build

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9919

### CODE REVIEW

File: rudder-agent/SPECS/rudder-agent.spec [hunk 1]
```diff
@@ -144,12 +144,12 @@ Requires: dmidecode
 
 ## 3 - SLES: No LMDB yet
 # Reference for suse_version : https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto
-%if 0%{?suse_version} && 0%{?suse_version} < 1140
+%if 0%{?suse_version} && 0%{?suse_version} < 1200
```
Comment (amousset) [inline rudder-agent/SPECS/rudder-agent.spec]: This breaks https://github.com/Normation/rudder-packages/pull/1005/files

File: rudder-agent/SPECS/rudder-agent.spec [hunk 2]
```diff
@@ -186,11 +186,10 @@ Requires: dmidecode
...
 ## 3 - SLES: Bundled for pre-sles11 oses
 ##
-### SLES 11 OSes come with OpenSSL 0.9.8h,
-### which is recent enough.
-### SLES12 has no sles_version defined, but openssl is ok too
+### SLES 11 OSes come with OpenSSL 0.9.8h which is recent enough.
 ##
-%if 0%{?sles_version} && 0%{?sles_version} < 11
+# Reference for suse_version : https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto
+%if 0%{?suse_version} && 0%{?suse_version} < 1200
```
Comment (amousset) [inline rudder-agent/SPECS/rudder-agent.spec]: We probably want to use system openssl onSLES11

File: rudder-agent/SPECS/rudder-agent.spec (hunk 1 above)
Comment (peckpeck) [inline rudder-agent/SPECS/rudder-agent.spec]: this is < 1200 (strictly inferior to 12) the old value was inferior to 1315

**Comment by peckpeck** [comment]
Commit modified

File: rudder-agent/SPECS/rudder-agent.spec (hunk 1 above)
Comment (amousset) [inline rudder-agent/SPECS/rudder-agent.spec]: Yes but it's not what we want for opensuse 11.4 at least

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
ping @amousset

File: rsyslog-suse/SPECS/rsyslog.spec [hunk 3]
```diff
@@ -145,7 +145,8 @@ dos2unix doc/*.html
 
 %build
 export CFLAGS="$RPM_OPT_FLAGS -fno-strict-aliasing -W -Wall"
-%if 0%{?suse_version} > 1000 && 0%{?suse_version} < 1030
```
Comment (amousset) [inline rsyslog-suse/SPECS/rsyslog.spec]: < 1030 does not match SELS 10 SP3, this changes the behavior on it (which is IIRC our only supported SLES 10). Is it intended?

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**